// @fullcalendar/angular/esm5/lib/fullcalendar.module.js
import { __decorate, __read, __spread } from "tslib";
import { NgModule } from '@angular/core';
import { globalPlugins } from '@fullcalendar/core';
import { FullCalendarComponent } from './fullcalendar.component';
import * as ɵngcc0 from '@angular/core';
var FullCalendarModule = /** @class */ (function () {
    function FullCalendarModule() {
    }
    FullCalendarModule.registerPlugins = function (plugins) {
        globalPlugins.push.apply(globalPlugins, __spread(plugins));
    };
FullCalendarModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: FullCalendarModule });
FullCalendarModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function FullCalendarModule_Factory(t) { return new (t || FullCalendarModule)(); }, imports: [[]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(FullCalendarModule, { declarations: function () { return [FullCalendarComponent]; }, exports: function () { return [FullCalendarComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(FullCalendarModule, [{
        type: NgModule,
        args: [{
                declarations: [FullCalendarComponent],
                imports: [],
                exports: [FullCalendarComponent]
            }]
    }], function () { return []; }, null); })();
    return FullCalendarModule;
}());
export { FullCalendarModule };
