// @fullcalendar/angular/esm2015/lib/fullcalendar.module.js
import { __decorate } from "tslib";
import { NgModule } from '@angular/core';
import { globalPlugins } from '@fullcalendar/core';
import { FullCalendarComponent } from './fullcalendar.component';
import * as ɵngcc0 from '@angular/core';
let FullCalendarModule = class FullCalendarModule {
    static registerPlugins(plugins) {
        globalPlugins.push(...plugins);
    }
};
FullCalendarModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: FullCalendarModule });
FullCalendarModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function FullCalendarModule_Factory(t) { return new (t || FullCalendarModule)(); }, imports: [[]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(FullCalendarModule, { declarations: function () { return [FullCalendarComponent]; }, exports: function () { return [FullCalendarComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(FullCalendarModule, [{
        type: NgModule,
        args: [{
                declarations: [FullCalendarComponent],
                imports: [],
                exports: [FullCalendarComponent]
            }]
    }], null, null); })();
export { FullCalendarModule };
